webapp/components/test-file-results.js

```diff
@@ -80,7 +83,65 @@ class TestFileResults extends TestRunsUIQuery(
   }
 
   static get observers() {
-    return ['fetchTestFile(path, testRuns)'];
+    return ['loadData(path, testRuns, structuredSearch)'];
+  }
+
+  async loadData(path, testRuns, structuredSearch) {
+    // Run a search query, including subtests, as well as fetching the results file.
+    let [searchResults, resultsTable] = await Promise.all([
+      this.fetchSearchResults(path, testRuns, structuredSearch),
+      this.fetchTestFile(path, testRuns),
+    ]);
+
+    if (resultsTable && searchResults) {
+      const test = searchResults.results.find(r => r.test === path);
+      if (test) {
+        let subtests = new Set(test.subtests);
+        let [first, ...others] = resultsTable;
+        const matches = others.filter(t => subtests.has(t.name));
+        resultsTable = [first, ...matches];
+      }
+    }
+    this.resultsTable = resultsTable;
+  }
+
+  async fetchSearchResults(path, testRuns, structuredSearch) {
+    if (!testRuns || !testRuns.length || !this.structuredQueries || !structuredSearch) {
+      return;
+    }
+
+    // Combine the query with " and [path]".
+    let q = {
+      and: [
+        {pattern: path},
+        structuredSearch,
+      ]
+    };
+
+    let url = new URL('/api/search', window.location);
+    url.searchParams.set('subtests', '');
+    const fetchOpts = {
+      method: 'POST',
+      body: JSON.stringify({
+        run_ids: testRuns.map(r => r.id),
+        query: q,
+      }),
+    };
+    return await this.retry(
+      async() => {
+        const r = await window.fetch(url, fetchOpts);
+        if (!r.ok) {
+          if (fetchOpts.method === 'POST' && r.status === 422) {
```

Comment:
Why the `fetchOpts.method === 'POST'` part of the condition? Nothing can have changed it at this point. Should it have been `r.method === 'POST'`?

Reply:
Copy-paste from a code branch that actually has both GET/POST.
For interop, we can't run a server-side search with GET.